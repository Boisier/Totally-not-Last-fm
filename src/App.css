.App
{
  display: flex;
  height: 100%;

  justify-content: center;
  flex-direction: column;

  text-align: center;
}
