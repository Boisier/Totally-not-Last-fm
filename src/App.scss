$pink: #ffcdd2;
$grey: #f8f9f9;
$yellow: #a3e4d7;

.App
{
  display: block;
  position: relative;

  height: 100%;
  width: 100%;
  background-color: $grey;
}
