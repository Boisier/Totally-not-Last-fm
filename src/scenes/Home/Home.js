import React, { Component } from 'react'

import CurrentlyListening from './scenes/CurrentlyListening/CurrentlyListening'
import MultipleStatsPeriodes from './scenes/MultipleStatsPeriodes/MultipleStatsPeriodes'

export default class extends Component {
  render = () => (
    <section className="home-page">
      <CurrentlyListening />
      <MultipleStatsPeriodes />
    </section>
  )
}
