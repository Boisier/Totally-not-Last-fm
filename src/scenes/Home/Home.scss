@import 'scenes/StatsPeriodesNavBar/StatsPeriodesNavBar';
@import 'scenes/StatsSection/StatsSection';
