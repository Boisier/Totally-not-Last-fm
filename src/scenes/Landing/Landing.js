import React, { Component } from 'react'

import WelcomeScreen from './components/WelcomeScreen/WelcomeScreen'
import SignUpForm from './components/SignUpForm/SignUpForm'
import SignInForm from './components/SignInForm/SignInForm'

import './Landing.scss'

export default class extends Component {
  constructor (props) {
    super(props)
    this.state = {
      showWelcomeScreen: true,
      showSignInForm: false,
      showSignUpForm: false,

      email: ''
    }
  }

  verifyEmail = (email) => {
    console.log('checking ' + email + ' ...')

    // TODO:Check email status with server

    // Email unknown, display signUp form
    this.setState({
      email: email
    })
    this.showSignUp()
  }

  showLanding = () => {
    this.setState({
      showWelcomeScreen: true,
      showSignInForm: false,
      showSignUpForm: false
    })
  }

  showSignUp = () => {
    this.setState({
      showWelcomeScreen: false,
      showSignInForm: false,
      showSignUpForm: true
    })
  }

  showSignIn = () => {
    this.setState({
      showWelcomeScreen: false,
      showSignInForm: true,
      showSignUpForm: false
    })
  }

  render = () => {
    const topPartClass = this.state.showSignUpForm || this.state.showSignInForm ? 'expand' : ''
    return (
      <main id="landing-page">
        <section className={'top-part ' + topPartClass}>
          <SignUpForm
            email={this.state.email}
            display={this.state.showSignUpForm}
            showLanding={this.showLanding}
            showSignUp={this.showSignUp}
            showSignIn={this.showSignIn} />
          <SignInForm
            email={this.state.email}
            display={this.state.showSignInForm}
            showLanding={this.showLanding}
            showSignUp={this.showSignUp}
            showSignIn={this.showSignIn} />
        </section>
        <WelcomeScreen
          verifyEmail={this.verifyEmail}
          display={this.state.showWelcomeScreen}/>
      </main>
    )
  }
}
