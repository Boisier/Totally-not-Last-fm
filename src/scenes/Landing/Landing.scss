$pink: #ffcdd2;
$grey: #f8f9f9;
$yellow: #a3e4d7;

#landing-page
{

  @import url('https://fonts.googleapis.com/css?family=Rubik:300,300i,400,400i,500,500i,700,700i,900,900i');

  font-family:'Rubik', sans-serif;
  width: 100%;
  height: 100%;

  overflow: hidden;
  /*
  background-image: url('https://scontent-cdg2-1.xx.fbcdn.net/v/t1.15752-9/31437460_1911799875817421_1060852300727189504_n.jpg?_nc_cat=0&oh=981dbb6d4387a9cb2180c2c962d4bfb7&oe=5B518C18');
  background-size: cover;
  background-position: bottom;
  background-blend-mode: soft-light;
  */

  background-image: url('http://4.bp.blogspot.com/_KckYYvB7Ge0/TJmFTb9XkhI/AAAAAAAAAaw/rjtOQ9I5EDA/s1600/headphones-.jpg'), url('https://i.pinimg.com/originals/96/aa/9e/96aa9e2d8ebdc6b0c66cd93a1ec56781.jpg');
  background-size: 30%, 30%;
  background-position:top right, bottom left;
  background-repeat: no-repeat;
  background-blend-mode: color-burn;

  //background-image: linear-gradient(to top right,  #fce4ec 0%, $pink 100%);
  background-color: $pink;
  .top-part
  {
    position: relative;
    width: 100%;
    height: 50%;

    transition-duration: .25s;
    transition-timing-function: ease-in-out;
  }
}

#welcome-screen
{
  transition-duration: .25s;
  transition-timing-function: ease-in-out;

  &.hide
  {
    margin-top: 100%;
  }

  width: 100%;
  height: 50%;

  position: relative;

  background-color: $grey;
  color: #fff;

  $side-margin: 5rem;

  h6
  {
    color: red;
    font-weight:500;
    font-size:2em;
  }

  .caption
  {
    display: block;
    margin: 0;
    padding: 0;

    position: absolute;
    top: -8rem;
    left: $side-margin;

    text-transform: uppercase;
    font-size: 1.25em;
  }

  .title
  {
    display: block;
    margin: 0;
    padding: 0;
    font-weight: 900;
    font-style: italic;
    position: absolute;
    text-transform: uppercase;
    font-size: 5em;
    left: $side-margin;

    &.line-1
    {
      color:$grey;
      top: -5.5rem;
      text-shadow: -3px 3px #ff0000;
    }

    &.line-2
    {
      color: #ff0000;
      top: .1rem;
    }
  }

  .access-form
  {
    display: block;
    width: 30rem;

    position: absolute;
    right: $side-margin;

    .field-input
    {
      position: absolute;
      top: -2.3rem;
    }

    input[type=button]
    {
      font-family:'Rubik';
      display: block;
      width: 15rem;
      height: 3rem;
      color: #fff;
      position: absolute;
      top: 1.7rem;
      right: 10px;

      border: 0;
      background-color: red;

      text-transform: uppercase;
      font-size: 0.8rem;
      font-weight:500;
      cursor:pointer;

      &:hover
      {
        color: #FFF;
        background-color: red;
      }
    }
  }
}
