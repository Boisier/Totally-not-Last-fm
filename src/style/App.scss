// Main SCSS Stylesheet

// Globals
@import 'variables';
@import 'fonts';
@import 'helpers';

// Containers
@import '../containers/**/*.scss';

// Scenes
@import '../scenes/**/*.scss';

/*
  APP
 */
.App
{
  display: block;
  position: relative;

  height: 100%;
  width: 100%;
  background-color: $grey;
}
