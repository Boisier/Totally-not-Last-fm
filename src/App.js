// React
import React, { Component } from 'react'
import { Switch, Route } from 'react-router-dom'

// Scene dependencies
import './App.scss'

// Scene componentts
import Header from './scenes/Header/Header'
import Structure from './scenes/Structure/Structure'
import Footer from './scenes/Footer/Footer'

import Home from './scenes/Home/Home'

export default class extends Component {
  render = () => (
    <section className="App">
      <Header/>
      <Structure>
        <Switch>
          <Route path='/' component={Home}/>
        </Switch>
      </Structure>
      <Footer />
    </section>
  )
}
