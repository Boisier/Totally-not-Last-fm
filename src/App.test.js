import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

import App from './App'
import Landing from './scenes/Landing/Landing'
import Home from './scenes/Home/Home'

const setup = propOverrides => {
  const props = Object.assign({
    url: '/',
    loggedIn: true
  }, propOverrides)

  const wrapper = mount(
    <MemoryRouter initialEntries={[props.url]}>
      <App isConnected={props.loggedIn}/>
    </MemoryRouter>
  )

  return {
    wrapper,
    landing: wrapper.find(Landing),
    home: wrapper.find(Home)
  }
}

describe('App display the correct page', () => {
  describe('When not connected', () => {
    it('/ => Landing page', () => {
      const { landing } = setup({loggedIn: false})
      expect(landing.length).toBe(1)
    })
  })
  describe('When connected', () => {
    it('/ => Home page', () => {
      const { home } = setup({url: '/'})
      expect(home.length).toBe(1)
    })
  })
})
