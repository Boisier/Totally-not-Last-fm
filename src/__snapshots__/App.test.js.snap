// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correct page based on URL 1`] = `
<section
  className="App"
>
  <header>
    <h1>
      Totally not Last fm online
    </h1>
  </header>
  <section
    className="home-page"
  >
    <span>
      CurrentlyListening
    </span>
    <section
      className="multiple-stats-periods"
    >
      <span>
        Nav bar for stats periods
      </span>
      <section
        className="multiple-stats-periods-container"
      >
        <span>
          StateSection
        </span>
        <span>
          StateSection
        </span>
        <span>
          StateSection
        </span>
        <span>
          StateSection
        </span>
      </section>
    </section>
  </section>
  <footer>
    <span>
      FOOTER
    </span>
  </footer>
</section>
`;
